protocol/0072-SPPW-spam-protection-PoW.md

```diff
@@ -12,46 +12,49 @@ spam.pow.increaseDifficulty = 0 (range: 0/1)
+      1. If this number is less than or equal to `spam.pow.numberOfTxPerBlock` the hash must start with `spam.pow.difficulty` zeros.
+      2. If this number is greater than `spam.pow.numberOfTxPerBlock` the hash must start with `spam.pow.difficulty` zeros plus one additional for each `spam.pow.numberOfTxPerBlock` sized batch of transactions beyond the limit (e.g. if `spam.pow.numberOfTxPerBlock` is 10 and `spam.pow.difficulty` is 2 the 11th - 20th transactions will require 2 zeros, the 21st-30th 3 zeros, the 31st-40th 4 zeros and so on).
+3. The user then attaches the PoW to a transaction for a block which will be created within `spam.pow.numberOfPastBlocks` blocks of the one used for PoW hash generation, and sends it off together with `x` and `H(b)`.
  
 The validators verify that
 - `H(b)` is the correct hash of a past block
 - That block is no more than `spam.pow.numberOfPastBlocks` in the past.
   - This check is primarily done by the leader (i.e., block creator). On agreeing on a new block, all parties check their mempool for now outdated transactions and purge them.
-- The hash is computed correctly and begins with `spam.pow.difficulty` zeroes
+- The hash is computed correctly and begins with `spam.pow.difficulty` zeros, or `spam.pow.difficulty` + n zeros if the validator has seen `spam.pow.numberOfTxPerBlock` * n blocks transactions from this party within the same block prior to this one.
+- Note that `spam.pow.numberOfTxPerBlock` is counted against the block used for PoW, not the present block. This means that a party could submit more than `spam.pow.numberOfTxPerBlock` transactions to a single created block at only `spam.pow.difficulty` by using multiple historic blocks all within `spam.pow.numberOfPastBlocks`.
    
 Furthermore, the validators check that:
 - The same identifier has not been used for another transaction from a previously committed block. If the same identifier is used for 
   several transactions in the same block, these transactions need to be removed during post-processing, and the initiator blocked as a 
   spammer.
-- The same block has not been used for more than `spam.pow.numberOfTxPerBlock` transactions per spam difficulty level 
+- The same block has not been used for more than `spam.pow.numberOfTxPerBlock` transactions by the same party per spam difficulty level 
   (i.e., if `spam.pow.increaseDifficulty` is `> 1`, the same block can be used for more transactions if the PoW accordingly increases in difficulty).
  
  Violations of the latter rules cannot lead to a transaction being removed, as different validators have a different view on 
- this; however, they can be verified post-agreement, and the offending vega-key can be banished for 4 epochs.
+ this; however, they can be verified post-agreement, and the offending vega-key can be banished for the maximum of 1/48 of an Epoch (30 minutes if an epoch is 1 day) or 30 seconds.
```

Comment:
maybe better as:
can be banished for the duration of 1/48 of an Epoch with a minimum duration of 30 seconds. E.g. if the epoch duration is 1 day, then the ban period is 30 minutes. If however the epoch is 10 seconds, then the ban period is 30 seconds.

Reply:
Sounds good